filer/utils/filer_easy_thumbnails.py

```diff
@@ -19,58 +18,41 @@ def thumbnail_to_original_filename(thumbnail_name):
 
 
 class ThumbnailerNameMixin:
-    thumbnail_basedir = ''
-    thumbnail_subdir = ''
-    thumbnail_prefix = ''
+    thumbnail_basedir = ""
+    thumbnail_subdir = ""
+    thumbnail_prefix = ""
 
     def get_thumbnail_name(self, thumbnail_options, transparent=False):
         """
-        A version of ``Thumbnailer.get_thumbnail_name`` that produces a
-        reproducible thumbnail name that can be converted back to the original
-        filename.
+        Get thumbnail name using easy-thumbnails pattern.
+        For public files: Uses configurable naming via THUMBNAIL_NAMER
+        For private files: Uses easy-thumbnails default naming pattern regardless of THUMBNAIL_NAMER
         """
-        path, source_filename = os.path.split(self.name)
-        source_extension = os.path.splitext(source_filename)[1][1:].lower()
-        preserve_extensions = self.thumbnail_preserve_extensions
-        if preserve_extensions is True or source_extension == 'svg' or \
-                isinstance(preserve_extensions, (list, tuple)) and source_extension in preserve_extensions:
-            extension = source_extension
-        elif transparent:
-            extension = self.thumbnail_transparency_extension
-        else:
-            extension = self.thumbnail_extension
-        extension = extension or 'jpg'
-
-        thumbnail_options = thumbnail_options.copy()
-        size = tuple(thumbnail_options.pop('size'))
-        initial_opts = ['{}x{}'.format(*size)]
-        quality = thumbnail_options.pop('quality', self.thumbnail_quality)
-        if extension == 'jpg':
-            initial_opts.append(f'q{quality}')
-        elif extension == 'svg':
-            thumbnail_options.pop('subsampling', None)
-            thumbnail_options.pop('upscale', None)
-
-        opts = list(thumbnail_options.items())
-        opts.sort()   # Sort the options so the file name is consistent.
-        opts = ['{}'.format(v is not True and f'{k}-{v}' or k)
-                for k, v in opts if v]
-        all_opts = '_'.join(initial_opts + opts)
+        is_public = False
+        if hasattr(self, "thumbnail_storage"):
+            from filer.storage import PrivateFileSystemStorage
+            is_public = not isinstance(self.thumbnail_storage, PrivateFileSystemStorage)
 
-        basedir = self.thumbnail_basedir
-        subdir = self.thumbnail_subdir
-
-        # make sure our magic delimiter is not used in all_opts
-        all_opts = all_opts.replace('__', '_')
-        filename = f'{source_filename}__{all_opts}.{extension}'
+        path, source_filename = os.path.split(self.name)
+        thumbnail_name = super(ThumbnailerNameMixin, self).get_thumbnail_name(
```

Comment:
I fear this call will look up the thumbnail namer setting and call that function. I see two ways to avoid that:
* Patch `self` to use the default thumbnail namer (e.g., using a context manager) if the file is not public
* Copy the current logic into filer

Reply:
@fsbraun My idea behind this was that we get all their optimizations and future improvements of easy thumbnails. This way we modify the returned value for private files (as verified in the tests).